com.microsoft.mrtk.graphicstools.unity/Runtime/MeshOutline/MeshOutline.cs

```diff
@@ -238,5 +240,40 @@ private static int GetMaxRenderQueue(Material[] materials)
 
             return max;
         }
+
+        /// <summary>
+        /// Creates and adds a SubMeshDescriptor which covers the entire mesh.
+        /// This ensures that the outline material is applied to the entire mesh.
+        /// </summary>
+        private void HandleMultipleSubMeshes()
+        {
+            Mesh mesh = null;
+
+            if (baseRenderer is SkinnedMeshRenderer skinnedMeshRenderer)
+            {
+                mesh = skinnedMeshRenderer.sharedMesh;
+            }
+            else if (baseRenderer is MeshRenderer meshRenderer)
+            {
+                mesh = meshRenderer.GetComponent<MeshFilter>().sharedMesh;
+            }
+
+            if (mesh != null)
+            {
+                var meshSubMeshCount = mesh.subMeshCount;
+                if (meshSubMeshCount > 1)
+                {
+                    var subMeshes = new SubMeshDescriptor[meshSubMeshCount + 1];
+                    for (int i = 0; i < meshSubMeshCount; i++)
+                    {
+                        var subMesh = mesh.GetSubMesh(i);
+                        subMeshes[i] = subMesh;
+                    }
+                    var lastMesh = subMeshes[meshSubMeshCount - 1];
+                    subMeshes[meshSubMeshCount] = new SubMeshDescriptor(0, lastMesh.indexStart + lastMesh.indexCount);
+                    mesh.SetSubMeshes(subMeshes, MeshUpdateFlags.DontRecalculateBounds);
```

Comment:
This is great! And very elegant. I don't have one to test, but do you know if this handles models with more than 1 sub mesh?

Reply:
Also, in the future it might be nice to include a sample mesh with submeshes in the sample scene to help test this code path.